cloudinit/net/eni.py

```diff
@@ -54,7 +54,7 @@ def _iface_add_subnet(iface, subnet):
             continue
         if key == 'address':
             value = "%s/%s" % (subnet['address'], subnet['prefix'])
-        if value and key in valid_map:
+        if value and key in valid_map + NET_CONFIG_COMMANDS:
```

Comment:
This doesn't just allow 'pre-up' but all of the NET_CONFIG_COMMANDS, none of which are modeled in any of the various network config formats that cloud-init parses.

What input config is including pre-up commands inside the network config yaml?  Can you share what that looks like?


I wonder if instead you should write out hook files in /etc/network/if-{down,post-down,pre-up,up}.d/<my hook>; where in there you can filter on the iface name and other attributes that are set on the interface.